### PR TITLE
feat(core): bulk methods for deleteTaskResultById, addExtSource and r…

### PR DESCRIPTION
**Pull request by xflord**
…emoveExtSource

* new bulk methods were created and added to RPC+openapi

### CODE REVIEW

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/TasksManagerMethod.java [hunk 1]
```diff
@@ -294,6 +294,19 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 		}
 	},
 
+	/*#
+	 * Delete TaskResult by its IDs
+	 *
+	 * @param taskResultIds List<int> IDs of TaskResult to deleted
+	 */
+	getDeleteTaskResultsByIds {
```
Comment (Johaney-s) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/TasksManagerMethod.java:302]:
```suggestion
	deleteTaskResultsByIds {
```

File: perun-base/src/main/java/cz/metacentrum/perun/core/api/exceptions/FacilityMismatchException.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package cz.metacentrum.perun.core.api.exceptions;
+
+/**
+ * Exception thrown when wrong facility provided.
+ *
+ */
+public class FacilityMismatchException extends PerunException {
```
Comment (Johaney-s) [inline perun-base/src/main/java/cz/metacentrum/perun/core/api/exceptions/FacilityMismatchException.java:7]: might come in handy if this exception has facility as optional property for later usage

File: perun-openapi/openapi.yml [hunk 3]
```diff
@@ -8491,6 +8559,29 @@ paths:
...
+      parameters:
+        - $ref: '#/components/parameters/groupId'
+        - name: sourceIds
+          description: "list of ext source ids"
+          schema:
+            type: array
+            items:
+              $ref: '#/components/parameters/sourceId'
+          in: query
+          required: true
```
Comment (HejdaJakub) [inline perun-openapi/openapi.yml:8577]: You can shorten all these four methods for extsources by defining a list of ext sources between parameters, something like that:
```
    extSourceIds:
      name: "extSources[]"
      description: "list of extSource ids List<Integer>"
      in: query
      required: true
      schema:
        type: array
        items:
          type: integer
```
```suggestion
        - $ref: '#/components/parameters/extSourceIds'
```

**Review by HejdaJakub — CHANGES_REQUESTED** [review]

**Comment by Johaney-s** [comment]
i think we should add tests for the new methods as it's part of DoD...

**Review by HejdaJakub — APPROVED** [review]

**Review by Johaney-s — APPROVED** [review]

**Pull request merged**